src/Infrastructure/Services/OfficeService/OfficeService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Application.Common;
+using Application.DTOs.OfficeDTO;
+using AutoMapper;
+using Infrastructure.Context;
+using Domain.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Services.OfficeService
+{
+    public class OfficeService : IOfficeService
+    {
+
+        private readonly ApplicationDbContext _dataContext;
+        private readonly IMapper _mapper;
+
+        public OfficeService(ApplicationDbContext dataContext,IMapper mapper)
+        {
+            _dataContext = dataContext;
+            _mapper = mapper;
```

Comment:
please add a null check in a contractor if it's null you should throw an exception
and apply it to every contractor